builtin/logical/transit/path_cache_config.go

```diff
@@ -0,0 +1,124 @@
+package transit
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/helper/keysutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func (b *backend) pathCacheConfig() *framework.Path {
+	return &framework.Path{
+		Pattern: "cache-config",
+		Fields: map[string]*framework.FieldSchema{
+			"cache-type": &framework.FieldSchema{
+				Type:     framework.TypeString,
+				Required: true,
+				Description: `
+Type of cache to use. Currently "syncmap" and "lru" are supported.
+`,
+			},
+
+			"cache-size": &framework.FieldSchema{
+				Type: framework.TypeInt,
+				Description: `
+Size of cache for a cache type that accepts a size. This is required for cache types
+that accept a size and currently applies only to "lru" cache type.
+`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback:    b.pathCacheConfigRead,
+				Summary:     "",
+				Description: "",
+			},
+
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:    b.pathCacheConfigWrite,
+				Summary:     "",
+				Description: "",
+			},
+
+			logical.CreateOperation: &framework.PathOperation{
+				Callback:    b.pathCacheConfigWrite,
+				Summary:     "",
+				Description: "",
+			},
+		},
+
+		HelpSynopsis:    pathCacheConfigHelpSyn,
+		HelpDescription: pathCacheConfigHelpDesc,
+	}
+}
+
+func (b *backend) pathCacheConfigWrite(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	// get target cacheType
+	cacheTypeStr := d.Get("cache-type").(string)
+	cacheSize := d.Get("cache-size").(int)
+	var cacheType keysutil.CacheType
+	switch cacheTypeStr {
+	case "syncmap":
+		cacheType = keysutil.SyncMap
+	case "lru":
+		cacheType = keysutil.LRU
+	default:
+		cacheType = keysutil.NotImplemented
+	}
+
+	// err if the requested cacheType has not been implemented
+	if cacheType == keysutil.NotImplemented {
+		return logical.ErrorResponse(fmt.Sprintf("unknown cache-type %q", cacheTypeStr)), logical.ErrInvalidRequest
+	}
+
+	// err if cacheType is lru but no cache-size was specified
+	if cacheType == keysutil.LRU && cacheSize <= 0 {
+		return logical.ErrorResponse("for lru cache-type, cache-size must be specified and be greater than zero"), logical.ErrInvalidRequest
+	}
+
+	// change the cache type
+	if cacheType == keysutil.SyncMap {
```

Comment:
Call convert if the cache type is different from the current one since it will reset the cache.